pages/previewModel.js

```diff
@@ -86,11 +115,20 @@ function loadModel(gltf, resetCamera) {
 
         viewer._model = model;
         viewer.onLoad.raiseEvent(model);
-    }).otherwise(function(e){
+    }).otherwise(function(e) {
         window.alert('Error: ' + e);
     });
 }
 
-var gltf = JSON.parse(document.getElementById('glTF').textContent);
+var gltfFileName = document.getElementById('gltfFileName').textContent;
+var gltfRootPath = document.getElementById('gltfRootPath').textContent;
 
-loadModel(gltf, true);
+try {
+    var gltfContent = JSON.parse(document.getElementById('gltf').textContent);
+    loadModelFromContent(gltfContent, gltfRootPath, true);
+}
+catch (ex) {
+    // If the glTF content is missing or not valid JSON, then try to load the
+    // model directly from the glTF file.
+    loadModelFromFile(gltfFileName, gltfRootPath, true);
```

Comment:
Ah, this is an important heads-up.  There are a couple of workarounds I can think of.

* VSCode allows the extension to register custom protocols, and this extension has 2 registered already.  We could register a 3rd (or a modification to the preview one) that returns the contents of the selected editor, with the relative paths corrected.  In theory, the URL to the custom protocol could then be passed into the various glTF loaders.

* Alternately, this file could call `new Blob(...)` on the contents of the glTF that were injected into a div, and get a Blob URI reference to those contents.  That reference acts as a URL that can be passed into the various loaders.